simplestore/src/main/java/com/uber/simplestore/impl/AtomicFile.java

```diff
@@ -105,6 +105,22 @@ public FileOutputStream startWrite() throws IOException {
     }
   }
 
+  private void resolveParentFolder(FileNotFoundException rawError) throws IOException {
```

Comment:
Those are minimum changes without introducing actual fix but to collect more information.